<launch>
  <arg name="no_static_map" default="true"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <rosparam file="$(find jackal_helper)/configs/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find jackal_helper)/configs/params/costmap_common_params.yaml" command="load" ns="local_costmap" />

    <rosparam file="$(find jackal_helper)/configs/params/odom_nav_params/global_costmap_params.yaml" command="load" />

    <rosparam file="$(find jackal_helper)/configs/params/odom_nav_params/local_costmap_params.yaml" command="load" />

    <param name="base_local_planner" value="hlsd_local_planner/HLSDPlanner" />

    <param name="base_global_planner" type="string" value="navfn/NavfnROS" />

    <param name="HLSDPlanner/model_path" value="$(find hlsd_local_planner)/models/hlsd_model.onnx" />
    <param name="HLSDPlanner/max_v" value="2.0" />
    <param name="HLSDPlanner/max_w" value="2.0" />
    <param name="HLSDPlanner/lidar_max_range" value="5.0" />
    <param name="HLSDPlanner/input_lidar_dim" value="60" />
    <param name="HLSDPlanner/goal_lookahead_dist" value="1.5" />

    <rosparam file="$(find jackal_helper)/configs/params/move_base_params.yaml" command="load" />

    <remap from="odom" to="odometry/filtered" />
    <remap from="cmd_vel" to="cmd_vel" />
  </node>
</launch>
